### PR TITLE
Refactor bioinfo tools version parsing

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Added:
- Sentieon version to bioinformatic tool version parsing #685 

Changed:
- Changed the way conda and non-conda bioinfo tools version as parsed

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by hassanfa** [comment]
ascatNgs and Sentieon are now included:
image

File: BALSAMIC/containers/sentieon/sentieon.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+- sentieon=20201002
```
Comment (ashwini06) [inline BALSAMIC/containers/sentieon/sentieon.yaml:1]:
```suggestion
- sentieon=20201002
```
```suggestion
- sentieon=202010.02
```

File: BALSAMIC/containers/ascatngs/ascatngs.yaml [hunk 2]
```diff
@@ -1,5 +1 @@
-channels:
-  - defaults
-
-dependencies:
-  - bioconda::ascat=4.5.0
+- ascatNgs=4.4.1
```
Comment (ashwini06) [inline BALSAMIC/containers/ascatngs/ascatngs.yaml:1]: Please check the differences in ascat version. Latest release is [4.5](https://github.com/cancerit/ascatNgs/tags)

File: BALSAMIC/utils/cli.py [hunk 3]
```diff
@@ -435,31 +435,90 @@ def get_panel_chrom(panel_bed) -> list:
     return {s.split("\t")[0] for s in lines}
 
 
+def bioinfo_tool_version_non_conda(packages: dict,
```
Comment (ashwini06) [inline BALSAMIC/utils/cli.py:438]: Cool 👍

File: tests/utils/test_utils.py [hunk 4]
```diff
@@ -149,15 +149,18 @@ def test_get_reference_output_files():
     assert "Homo_sapiens_assembly38.fasta" in fasta_files
 
 
-def test_get_bioinfo_tools_list():
+def test_get_bioinfo_tools_version():
```
Comment (ashwini06) [inline tests/utils/test_utils.py:152]: This test is for testing bioinfo_tool from conda env. Do we need one more test function, for testing bioinfo_tool from nonconda env?

**Review by ashwini06 — COMMENTED** [review]
Nice one 👍   Few minor comments

**Comment by ashwini06** [comment]
> ascatNgs and Sentieon are now included:
> image

Why the sentieon version is not displayed?

**Comment by hassanfa** [comment]
I forgot to paste Sentieon in config:
image

File: BALSAMIC/containers/ascatngs/ascatngs.yaml (hunk 2 above)
Comment (hassanfa) [reply BALSAMIC/containers/ascatngs/ascatngs.yaml:1]: You're right. I'll fix it.

File: BALSAMIC/containers/sentieon/sentieon.yaml (hunk 1 above)
Comment (hassanfa) [reply BALSAMIC/containers/sentieon/sentieon.yaml:1]: fixed!

File: tests/utils/test_utils.py (hunk 4 above)
Comment (hassanfa) [reply tests/utils/test_utils.py:152]: Proper way should've been to write three tests... this one will pick up for all three functions... Not a unittest though :-p

**Review by khurrammaqbool — APPROVED** [review]
Nicely re-arranged handling of conda and submodule packages. 👍

File: BALSAMIC/containers/ascatngs/ascatngs.yaml [hunk 5]
```diff
@@ -1,5 +1,2 @@
-channels:
-  - defaults
-
-dependencies:
-  - bioconda::ascat=4.5.0
+- ascatNgs=4.5.0
+- alleleCount
```
[the remark this answers is not in the document]
Comment (khurrammaqbool) [reply BALSAMIC/containers/ascatngs/ascatngs.yaml:2]: ascatngs version here is 4.5.0, but .json above it shows 4.4.1

[the remark this answers is not in the document]
Comment (hassanfa) [reply BALSAMIC/containers/ascatngs/ascatngs.yaml:2]: I'll update json.

I was testing, and pushed by mistake. Good catch!

File: BALSAMIC/utils/cli.py [hunk 6]
```diff
@@ -435,31 +435,87 @@ def get_panel_chrom(panel_bed) -> list:
...
```
[the remark this answers is not in the document]
Comment (hassanfa) [reply BALSAMIC/utils/cli.py:438]: Probably a good idea, but no need for now. All non-conda softwares will be parsed in this function.

File: BALSAMIC/utils/cli.py [hunk 7]
```diff
@@ -435,31 +435,87 @@ def get_panel_chrom(panel_bed) -> list:
...
+        version = "=".join(p.split("=")[1:])
+        if name not in bioinfo_tools:
+            continue
+        bioinfo_version[name] = list([version])
+
+    return bioinfo_version
+
+
+def bioinfo_tool_version_conda(packages: list, bioinfo_tools: dict, current_bioinfo_tool_version:dict) -> dict:
+    """Parses conda environment dictionary and extracts depdencies as version
```
Comment (ashwini06) [inline BALSAMIC/utils/cli.py:461]:
```suggestion
    """Parses conda environment dictionary and extracts dependencies as version
```

File: BALSAMIC/utils/cli.py [hunk 8]
```diff
@@ -435,31 +435,87 @@ def get_panel_chrom(panel_bed) -> list:
...
+
 def get_bioinfo_tools_version(bioinfo_tools: dict,
                               container_conda_env_path: os.PathLike) -> dict:
     """Parses the names and versions of bioinfo tools 
-    used by BALSAMIC from config YAML into a dict """
+    used by BALSAMIC from config YAML into a dict.
+
+    Args:
+        bioinfo_tools: dict. A dictionary with bioinfo tools as keys and container as value
+        container_conda_env_path: path. Path to all container and conda yaml there of
```
Comment (ashwini06) [inline BALSAMIC/utils/cli.py:496]:
```suggestion
        container_conda_env_path: path. Path to all container and conda yaml 
```

File: CHANGELOG.rst [hunk 9]
```diff
@@ -42,7 +43,8 @@ Changed:
 * Generalized ascat, delly, and manta result in workflow. #708
 * Generalized workflow to eliminate duplicate entries and code. #708
 * Split Tumor-Normal ``cnvkit batch`` into individual commands
->>>>>>> origin/develop
+* Changed the way conda and non-conda bioinfo tools version as parsed
```
Comment (ashwini06) [inline CHANGELOG.rst:46]:
```suggestion
* Changed the way conda and non-conda bioinfo tools version are parsed
```

**Review by ashwini06 — APPROVED** [review]

**Comment by ashwini06** [comment]
> Kudos, SonarCloud Quality Gate passed!    Quality Gate passed
> 
> [Bug](https://sonarcloud.io/project/issues?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=BUG)
> [Vulnerability](https://sonarcloud.io/project/issues?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=VULNERABILITY)
> [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=SECURITY_HOTSPOT)
> [Code Smell](https://sonarcloud.io/project/issues?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=Clinical-Genomics_BALSAMIC&pullRequest=713&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=Clinical-Genomics_BALSAMIC&pullRequest=713) No Coverage information
> [](https://sonarcloud.io/component_measures?id=Clinical-Genomics_BALSAMIC&pullRequest=713&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=Clinical-Genomics_BALSAMIC&pullRequest=713&metric=new_duplicated_lines_density&view=list)

@hassanfa : Please check code smell before merge

**Comment by hassanfa** [comment]
ascatNgs version: 
image


Sentieon version:
image

**Pull request merged**